### ISSUE TITLE
BS4 Startup issues

### ISSUE THREAD

**Issue by Nuytemans-Dieter** [Nuytemans-Dieter/BetterSleeping#257]
Startup errors have been reported:
On Java 11 and PaperSpigot 1.16.5:
```
[12:48:18 ERROR]: Error occurred while enabling BetterSleeping4 v4.0.0 (Is it up to date?)
java.lang.NoClassDefFoundError: javax/inject/Provider
at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
at java.lang.ClassLoader.defineClass(ClassLoader.java:1016) ~[?:?]
at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174) ~[?:?]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:178) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:216) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:122) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Paper-574]
at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
at com.google.inject.Scopes.<clinit>(Scopes.java:37) ~[?:?]
at com.google.inject.internal.InjectorShell$RootModule.configure(InjectorShell.java:340) ~[?:?]
at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:409) ~[?:?]
at com.google.inject.spi.Elements.getElements(Elements.java:108) ~[?:?]
at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:160) ~[?:?]
at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107) ~[?:?]
at com.google.inject.Guice.createInjector(Guice.java:87) ~[?:?]
at com.google.inject.Guice.createInjector(Guice.java:69) ~[?:?]
at com.google.inject.Guice.createInjector(Guice.java:59) ~[?:?]
at be.betterplugins.bettersleeping.BetterSleeping.startPlugin(BetterSleeping.java:71) ~[?:?]
at be.betterplugins.bettersleeping.BetterSleeping.onEnable(BetterSleeping.java:65) ~[?:?]
at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Paper-574]
at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:580) ~[patched_1.16.5.jar:git-Paper-574]
at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:280) ~[patched_1.16.5.jar:git-Paper-574]
at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055) ~[patched_1.16.5.jar:git-Paper-574]
at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289) ~[patched_1.16.5.jar:git-Paper-574]
at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException: javax.inject.Provider
at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:182) ~[patched_1.16.5.jar:git-Paper-574]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Paper-574]
at java.lang.ClassLoader.loadClass(ClassLoader.java:588) ~[?:?]
at java.lang.ClassLoader.loadClass(ClassLoader.java:521) ~[?:?]
... 30 more
[12:48:18 INFO]: [BetterSleeping4] Disabling BetterSleeping4 v4.0.0
```
PurPur 1.16.5 on Java 11
```
[00:03:21 INFO]: [BetterSleeping4] Enabling BetterSleeping4 v4.0.0
[00:03:21 ERROR]: Error occurred while enabling BetterSleeping4 v4.0.0 (Is it up to date?)
java.lang.NoClassDefFoundError: javax/inject/Provider
        at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
        at java.lang.ClassLoader.defineClass(ClassLoader.java:1017) ~[?:?]
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:179) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:216) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:123) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Purpur-1076]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        at com.google.inject.Scopes.<clinit>(Scopes.java:37) ~[?:?]
        at com.google.inject.internal.InjectorShell$RootModule.configure(InjectorShell.java:340) ~[?:?]
        at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:409) ~[?:?]
        at com.google.inject.spi.Elements.getElements(Elements.java:108) ~[?:?]
        at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:160) ~[?:?]
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:87) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:69) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:59) ~[?:?]
        at be.betterplugins.bettersleeping.BetterSleeping.startPlugin(BetterSleeping.java:71) ~[?:?]
        at be.betterplugins.bettersleeping.BetterSleeping.onEnable(BetterSleeping.java:65) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Purpur-1076]
        at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:593) ~[patched_1.16.5.jar:git-Purpur-1076]
        at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:293) ~[patched_1.16.5.jar:git-Purpur-1076]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1070) ~[patched_1.16.5.jar:git-Purpur-1076]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Purpur-1076]
        at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.lang.ClassNotFoundException: javax.inject.Provider
        at java.net.URLClassLoader.findClass(URLClassLoader.java:471) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:183) ~[patched_1.16.5.jar:git-Purpur-1076]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Purpur-1076]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:589) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:522) ~[?:?]
        ... 30 more
[00:03:21 INFO]: [BetterSleeping4] Disabling BetterSleeping4 v4.0.0
```

**Comment by Nuytemans-Dieter** [comment]
Another report:
```
[17:13:44 ERROR]: Error occurred while enabling BetterSleeping4 v4.0.0 (Is it up to date?)
java.lang.NoClassDefFoundError: javax/inject/Provider
        at java.lang.ClassLoader.defineClass1(Native Method) ~[?:?]
        at java.lang.ClassLoader.defineClass(ClassLoader.java:1010) ~[?:?]
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:150) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:179) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:216) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:123) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Purpur-1079]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:586) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
        at com.google.inject.Scopes.<clinit>(Scopes.java:37) ~[?:?]
        at com.google.inject.internal.InjectorShell$RootModule.configure(InjectorShell.java:340) ~[?:?]
        at com.google.inject.spi.Elements$RecordingBinder.install(Elements.java:409) ~[?:?]
        at com.google.inject.spi.Elements.getElements(Elements.java:108) ~[?:?]
        at com.google.inject.internal.InjectorShell$Builder.build(InjectorShell.java:160) ~[?:?]
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:107) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:87) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:69) ~[?:?]
        at com.google.inject.Guice.createInjector(Guice.java:59) ~[?:?]
        at be.betterplugins.bettersleeping.BetterSleeping.startPlugin(BetterSleeping.java:71) ~[?:?]
        at be.betterplugins.bettersleeping.BetterSleeping.onEnable(BetterSleeping.java:65) ~[?:?]
        at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415) ~[patched_1.16.5.jar:git-Purpur-1079]
        at net.minecraft.server.v1_16_R3.MinecraftServer.loadWorld(MinecraftServer.java:593) ~[patched_1.16.5.jar:git-Purpur-1079]
        at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:293) ~[patched_1.16.5.jar:git-Purpur-1079]
        at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1070) ~[patched_1.16.5.jar:git-Purpur-1079]
        at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:291) ~[patched_1.16.5.jar:git-Purpur-1079]
        at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.lang.ClassNotFoundException: javax.inject.Provider
        at java.net.URLClassLoader.findClass(URLClassLoader.java:433) ~[?:?]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:183) ~[patched_1.16.5.jar:git-Purpur-1079]
        at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:111) ~[patched_1.16.5.jar:git-Purpur-1079]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:586) ~[?:?]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:519) ~[?:?]
        ... 30 more
[17:13:44 INFO]: [BetterSleeping4] Disabling BetterSleeping4 v4.0.0
```

**Comment by Nuytemans-Dieter** [comment]
Should be fixed in cde918bbf403452aaab2e2dbb4a7ef599b28ec68
